lib/cisco_node_utils/vxlan_global.rb

```diff
@@ -0,0 +1,158 @@
+# VXLAN global provider class
+# Provides configuration of anycast gateways and duplicate host IP and
+# mac detection
+#
+# Alok Aggarwal, October 2015
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.join(File.dirname(__FILE__), 'node_util')
+
+module Cisco
+  # node_utils class for vxlan_global
+  class VxlanGlobal < NodeUtil
+    def feature_enable
+      config_set('vxlan_global', 'feature', state: '')
+    end
+
+    def feature_disable
+      config_set('vxlan_global', 'feature', state: 'no')
+    end
+
+    # Check current state of the configuration
+    def self.feature_enabled
+      feat = config_get('vxlan_global', 'feature')
+      return !(feat.nil? || feat.empty?)
+    rescue Cisco::CliError => e
+      # This cmd will syntax reject if feature is not
+      # enabled. Just catch the reject and return false.
+      return false if e.clierror =~ /Syntax error/
+      raise
+    end
+
+    # ----------
+    # PROPERTIES
+    # ----------
+
+    # dup-host-ip-addr-detection
+    def dup_host_ip_addr_detection
+      get_args = {}
+      match = config_get('vxlan_global', 'dup_host_ip_addr_detection', get_args)
+      if match.nil? || match.first.nil?
+        default_dup_host_ip_addr_detection
+      else
+        match.first.collect(&:to_i)
+      end
+    end
+
+    def dup_host_ip_addr_detection_host_moves
+      host_moves, _timeout = dup_host_ip_addr_detection
+      return default_dup_host_ip_addr_detection_host_moves if host_moves.nil?
+      host_moves
+    end
+
+    def dup_host_ip_addr_detection_timeout
+      _host_moves, timeout = dup_host_ip_addr_detection
+      return default_dup_host_ip_addr_detection_timeout if timeout.nil?
+      timeout
+    end
+
+    def dup_host_ip_addr_detection_set(state, host_moves, timeout)
```

Comment:
You're using state incorrectly here because you're expecting the callers to pass in 'no ' or '', which means they'll require some knowledge of the platform cli, which is bad if we're trying to abstract this away from the agents. They should be passing in true/false and then this method can set it appropriately to 'no ' or ''.